src/parachain/loans.ts

```diff
@@ -28,105 +40,425 @@ const MOCKDATA_LOAN_ASSET_KINT = {
      *
-     * @param accountId the account Id for which to get the collateral balance
-     * @param currency The currency to retrieve the collateral balance for
-     * @returns The current collateral amount
-     */
-    getCurrentCollateralBalance(accountId: AccountId, currency: CurrencyExt): Promise<MonetaryAmount<CurrencyExt>>;
-
-    /**
-     * Get the supply positions for given account.
-     * 
      * @param accountId the account Id for which to get supply positions
-     * @returns Array of supply positions of account.
+     * @returns Array of lend positions of account.
      */
     getLendPositionsOfAccount(accountId: AccountId): Promise<Array<LendPosition>>;
 
     /**
      * Get the borrow positions for given account.
-     * 
+     *
      * @param accountId the account Id for which to get borrow positions
      * @returns Array of borrow positions of account.
      */
     getBorrowPositionsOfAccount(accountId: AccountId): Promise<Array<BorrowPosition>>;
 
     /**
      * Get all loan assets.
-     * 
+     *
      * @returns Array of all assets that can be lent and borrowed.
      */
     getLoanAssets(): Promise<TickerToData<LoanAsset>>;
+
+    /**
+     * Get underlying currency of lend token id,
+     *
+     * @param lendTokenId Currency id of the lend token to get currency from
+     * @returns Underlying CurrencyExt for provided lend token
+     */
+    getUnderlyingCurrencyFromLendTokenId(lendTokenId: InterbtcPrimitivesCurrencyId): Promise<CurrencyExt>;
+
+    /**
+     * Get all lend token currencies.
+     *
+     * @returns Array of all LendToken currencies.
+     */
+    getLendTokens(): Promise<Array<LendToken>>;
+
+    /**
+     * Lend currency to protocol for borrowing.
+     *
+     * @param {AccountId} accountId Account that want to lend.
+     * @param {CurrencyExt} underlyingCurrency  Currency to lend.
+     * @param {MonetaryAmount<CurrencyExt>} amount Amount of currency to lend.
+     * @throws If there is not active market for `underlyingCurrency`.
+     * @throws If `amount` is exceeding available balance of account.
+     */
+    lend(accountId: AccountId, underlyingCurrency: CurrencyExt, amount: MonetaryAmount<CurrencyExt>): Promise<void>;
+
+    /**
+     * Withdraw previously lent currency from protocol.
+     *
+     * @param {CurrencyExt} underlyingCurrency Currency to witdhraw.
+     * @param {MonetaryAmount<CurrencyExt>} amount Amount of currency to withdraw.
+     * @throws If there is not active market for `underlyingCurrency`.
+     * @throws If `amount` is exceeding lent amount of account.
+     * @throws If `underlyingCurrency` is used as collateral and withdrawal of
+     * `amount` would bring account under collateral threshold.
+     * @throws If there is not enough of underlying currency currently
+     * available in the protocol.
+     */
+    withdraw(underlyingCurrency: CurrencyExt, amount: MonetaryAmount<CurrencyExt>): Promise<void>;
```

Comment:
this extrinsic has been removed since it's performed under-the-hood by the parachain